### PR TITLE
feat: use lowercase http method

### LINKED ISSUE #52: Make sure http verbs are of lower case.
`serverless.yml` allows capital verbs in `http` event. 

```yaml
  my-delete:
    name: my-delete-${self:provider.stage}
    description: Endpoint to delete
    handler: api.endpoints.DELETE
    events:
      - http: 
          method: DELETE
          path: blah/{id}
```

This will fail as only lower case verbs are supported by [openapi](https://swagger.io/docs/specification/paths-and-operations/#:~:text=OpenAPI%203.0%20supports%20get%2C%20post%2C%20put%2C%20patch%2C%20delete%2C%20head%2C%20options%2C%20and%20trace.).

### PR DESCRIPTION
**Pull request by bfaulk96**
fixes #52

### CODE REVIEW

**Comment by bfaulk96** [comment]
@desaintvincent Would you mind taking a look?

**Review by desaintvincent — APPROVED** [review]
great work! really clean
Few feedback:
1) it's pity to use methods so much. Some standalone function with a parameter would help to unit test and reduce that big file
2) (personal opinion here): It would make more sense to use CI for tests and lint than husky because:
- husky is local. Anyone can bypass it and push dirty code
- I do not like something to prevent me to push (wip code for exemple)
- CI prevent merge and not push itself

**Comment by bfaulk96** [comment]
@desaintvincent thank you for reviewing!

> it's pity to use methods so much. Some standalone function with a parameter would help to unit test and reduce that big file

I fully agree with you. I'm trying to chip away at some of the code smells piece-by-piece, since I joined in helping maintain this project well after it was initially written. I'm still trying to learn the ins and outs and make sure I don't break things moving them around 😄 

> (personal opinion here): It would make more sense to use CI for tests and lint than husky because:
> - husky is local. Anyone can bypass it and push dirty code
> - I do not like something to prevent me to push (wip code for exemple)
> - CI prevent merge and not push itself

On this note, I prefer defaulting to strict over loose. Users can always skip husky for WIP with `--no-verify` (as you mentioned in your first bullet point).. The husky scripts aren't to prevent bad practices, they're to help encourage them before the PR. We still block the PR on failed unit tests, but adding these things to push allows new contributors to potentially see their tests failing before it gets to GitHub.

I'd be open to further discussing removing unit tests from the push hook, but I do firmly believe the prettier/eslint stuff is beneficial as husky hooks

**Comment by desaintvincent** [comment]
Husky or not, I truly believe that lint should occur CI-side in order to prevent code quality to lower

**Comment by bfaulk96** [comment]
> Husky or not, I truly believe that lint should occur CI-side in order to prevent code quality to lower

Agreed!

File: tests/otherTests.test.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
...
-describe('empty tests', () => {
-    test('test', () => {
-        expect(true).toBe(true);
-    });
-    //TODO add more tests
-});
+'use strict';
+describe('empty tests', () => {
+  test('test', () => {
+    expect(true).toBe(true);
```
[the remark this answers is not in the document]
Comment (bfaulk96) [reply tests/otherTests.test.js]: Resolved all `.test` comments!

**Comment by bfaulk96** [comment]
@desaintvincent Could I ask ya for one more review? Looking to merge this, but want to get one more set of eyes on the new changes first

**Comment by desaintvincent** [comment]
@bfaulk96 seems ok!

**Pull request merged**